src/sql/workbench/contrib/dataExplorer/browser/dataExplorer.contribution.ts

```diff
@@ -35,13 +36,23 @@ configurationRegistry.registerConfiguration({
 	'title': localize('databaseConnections', "Database Connections"),
 	'type': 'object',
 	'properties': {
-		'datasource.connections': {
-			'description': localize('datasource.connections', "data source connections"),
+		[CONNECTIONS_CONFIG_KEY]: {
+			'description': localize(CONNECTIONS_CONFIG_KEY, "data source connections"),
 			'type': 'array'
 		},
-		'datasource.connectionGroups': {
-			'description': localize('datasource.connectionGroups', "data source groups"),
+		[GROUPS_CONFIG_KEY]: {
+			'description': localize(GROUPS_CONFIG_KEY, "data source groups"),
 			'type': 'array'
+		},
+		[CONNECTION_SORT_BY_CONFIG_KEY]: {
+			'type': 'string',
+			'enum': ['DateAdded', 'DisplayName'],
+			'enumDescriptions': [
+				localize('sortBy.DateAdded', 'Saved connections are sorted by the date they were added.'),
```

Comment:
`connections.sortBy.dateAdded` 
`connections.sortBy.displayName`

Reply:
updated. I also tried to eliminate string literals but `'connections.sortBy.' + ConnectionsSortBy.dateAdded` doesn't look any better than `connections.sortBy.dateAdded` so I just went with string literals here.

Reply:
Yeah string literals for the loc IDs are fine since those don't have to match anything specifically. It's more about making it so the ID is descriptive enough for what the string contains to avoid confusion.